### PR TITLE
[lcars/notanalog/limelight/daisy/thering/widviz/90sclk/widviztime/neonx] Add widget utils

### PR DESCRIPTION
**Pull request by thyttan**
I've converted the remaining clocks and widgets in #2196 to use widget utils.

I have not tested any of them and don't really intend to. If someone feels up for it please go ahead.

Especially widviz and widviztime I don't know if I've messed up or if they are ok. I can drop them from the PR if that's better.

For the others I feel it could be ok to merge them and if bugs come up we can fix them then?

### CODE REVIEW

File: apps/widviztime/widget.js [hunk 1]
```diff
@@ -1,30 +1,20 @@
...
 
     function reveal() {
         if (!Bangle.isLCDOn() || !saved) return;
-        for (var wd of WIDGETS) {
-            var o = saved.shift();
-            wd.draw = o.d;
-            wd.area = o.a;
-        }
+        widget_utils.show();
         Bangle.drawWidgets();
```
Comment (gfwilliams) [inline apps/widviztime/widget.js:18]: `Bangle.drawWidgets();` shouldn't be needed here, and I think you need `saved=null`

File: apps/widviztime/widget.js [hunk 2]
```diff
@@ -1,30 +1,20 @@
 (() => {
+    let widget_utils = require('widget_utils');
 
     var saved = null;
 
 
     function hide() {
         if (!Bangle.isLCDOn() || saved) return;
         saved = [];
```
Comment (gfwilliams) [inline apps/widviztime/widget.js:9]: I think you need saved=true here

File: apps/widviz/widget.js [hunk 3]
```diff
@@ -1,26 +1,19 @@
 (() => {
+  let widget_utils = require('widget_utils');
 
   var saved = null;
 
   function hide(){
     if (!Bangle.isLCDOn() || saved) return;
     saved = [];
```
Comment (gfwilliams) [inline apps/widviz/widget.js:8]: saved=true here instead

File: apps/widviz/widget.js [hunk 4]
```diff
@@ -1,26 +1,19 @@
...
 
   function reveal(){
     if (!Bangle.isLCDOn() || !saved) return;
-    for (var wd of WIDGETS) {
-      var o = saved.shift();
-      wd.draw = o.d;
-      wd.area = o.a;
-    }
+    widget_utils.show();
     Bangle.drawWidgets();
```
Comment (gfwilliams) [inline apps/widviz/widget.js:17]: You can remove drawWidgets but you need saved=false

File: apps/widviztime/widget.js (hunk 1 above)
Comment (thyttan) [reply apps/widviztime/widget.js:18]: null or false doesn't matter?

**Comment by thyttan** [comment]
Updated per suggestions.

File: apps/widviz/widget.js (hunk 4 above)
Comment (gfwilliams) [reply apps/widviz/widget.js]: It shouldn't, no... But using false everywhere is nicer

**Comment by thyttan** [comment]
>Thanks, but I don't think you quite updated the widgets properly... See the comments - GitHub's message thing is annoying as I left the comment for saved=true here instead next to  saved = []; - but it looks from the section posted above like I meant it for var saved = null;

Yeah I see, if I were a little more observant I could have worked that out, sorry :P

**Comment by gfwilliams** [comment]
Great - thanks!

**Pull request merged**